cli: TestPartialZip failed

cli.TestPartialZip [failed](https://mesolite.cluster.engflow.com/invocations/default/6fde597e-6831-4393-aa94-085d832405cf?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-25.2 @ [8b09c04432af7832135fb4085bbb02a20662a500](https://github.com/cockroachdb/cockroach/commits/8b09c04432af7832135fb4085bbb02a20662a500):


```
          [node 3] node status... writing JSON output: debug/nodes/3/status.json... done
          [node 3] using SQL connection URL: postgresql://...
          <dumping SQL tables>
          [node 3] requesting data for debug/nodes/3/details... received response... writing JSON output: debug/nodes/3/details.json... done
          [node 3] requesting data for debug/nodes/3/gossip... received response... writing JSON output: debug/nodes/3/gossip.json... done
          [node 3] requesting stacks... received response... writing binary output: debug/nodes/3/stacks.txt... done
          [node 3] requesting stacks with labels... received response... writing binary output: debug/nodes/3/stacks_with_labels.txt... done
          [node 3] requesting heap profile... received response... writing binary output: debug/nodes/3/heap.pprof... done
          [node 3] requesting engine stats... received response... writing binary output: debug/nodes/3/lsm.txt... done
          [node 3] requesting heap profile list... received response...
          [node 3] requesting heap profile list: last request failed: rpc error: ...
          [node 3] requesting heap profile list: creating error output: debug/nodes/3/heapprof.err.txt... done
          [node 3] requesting goroutine dump list... received response...
          [node 3] requesting goroutine dump list: last request failed: rpc error: ...
          [node 3] requesting goroutine dump list: creating error output: debug/nodes/3/goroutines.err.txt... done
          [node 3] requesting cpu profile list... received response...
          [node 3] requesting cpu profile list: last request failed: rpc error: ...
          [node 3] requesting cpu profile list: creating error output: debug/nodes/3/cpuprof.err.txt... done
          [node 3] requesting log files list... received response... done
          [node ?] ? log files found
          [node 3] requesting ranges... received response... done
          [node 3] writing ranges... writing JSON output: debug/nodes/3/ranges.json... done
          [cluster] pprof summary script... writing binary output: debug/pprof-summary.sh... done
          [cluster] hot range summary script... writing binary output: debug/hot-ranges.sh... done
          [cluster] tenant hot range summary script... writing binary output: debug/hot-ranges-tenant.sh... done
          [cluster] capture debug zip flags... writing binary output: debug/debug_zip_command_flags.txt... done
[debug --host=127.0.0.1:41053 --insecure=true zip --concurrency=1 --exclude-nodes=2 --cpu-profile-duration=0 --cluster-name=TestCluster-2W5wzRzDnc /dev/null]
    zip_test.go:770: 
          /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/testdata/zip/partial1_excluded:1:
          zip
        output didn't match expected:
          @@ -14,10 +14,14 @@
           [cluster] requesting tenant ranges: last request failed: rpc error: ...
           [cluster] requesting tenant ranges: creating error output: debug/tenant_ranges.err.txt... done
           [cluster] collecting the inflight traces for jobs... received response... done
           [node 1] node status... writing JSON output: debug/nodes/1/status.json... done
           [node 1] using SQL connection URL: postgresql://...
          +<dumping SQL tables>
          +[node 1] retrieving SQL data for crdb_internal.node_queries... writing output: debug/nodes/1/crdb_internal.node_queries.txt...NOTICE: No license is installed. Throttling will begin after 2026-03-21 08:07:40 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2026-03-21 08:07:40 +0000 UTC unless a license is installed before then.
          + done
           <dumping SQL tables>
           [node 1] requesting data for debug/nodes/1/details... received response... writing JSON output: debug/nodes/1/details.json... done
           [node 1] requesting data for debug/nodes/1/gossip... received response... writing JSON output: debug/nodes/1/gossip.json... done
           [node 1] requesting stacks... received response... writing binary output: debug/nodes/1/stacks.txt... done
           [node 1] requesting stacks with labels... received response... writing binary output: debug/nodes/1/stacks_with_labels.txt... done
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestPartialZip2379305014
    panic.go:635: -- test log scope end --
--- FAIL: TestPartialZip (29.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPartialZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61581

--- a/pkg/cli/testutils.go
+++ b/pkg/cli/testutils.go
@@ -364,14 +364,19 @@ func captureOutput(f func()) (out string, err error) {
 		if x := recover(); x != nil {
 			err = errors.Errorf("panic: %v", x)
 		}
+		// Strip license-related NOTICE messages that the SQL driver may inject
+		// into output. These appear nondeterministically and can split retrieval
+		// lines across multiple lines, breaking the <dumping SQL tables> collapsing
+		// below. Strip them early so retrieval lines remain intact.
+		out = regexp.MustCompile(`NOTICE: No license is installed[^\n]*\n?`).
+			ReplaceAllString(out, "")
 		// Replace any series of 'retrieving SQL data for ...' messages with a
 		// single '<dumping SQL tables>' message so that these tests are agnostic to
 		// both specific names and total number of system and internal tables that
 		// are exported. The regex matches the rest of the line after the prefix
 		// unless the line contains an uppercase E to avoid trimming "ERROR" message
-		// lines (but not NOTICE which is informational), which are expected
-		// (and tested) for certain tables.
-		out = regexp.MustCompile(`(.*retrieving SQL data for (([^E\n])*|.*NOTICE.*)\n)+`).
+		// lines, which are expected (and tested) for certain tables.
+		out = regexp.MustCompile(`(.*retrieving SQL data for ([^E\n])*\n)+`).
 			ReplaceAllString(out, "<dumping SQL tables>\n")
 	}()
 

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -724,7 +724,9 @@ func eraseNonDeterministicZipOutput(out string) string {
 	out = re.ReplaceAllString(out, `[node ?] ? execution traces found`)
 
 	// Remove license-related NOTICE messages that may appear intermittently.
-	re = regexp.MustCompile(`(?m)^NOTICE: No license is installed.*\n?`)
+	// Most NOTICE messages are stripped earlier in RunWithCapture, but this
+	// catches any that appear outside of SQL data retrieval lines.
+	re = regexp.MustCompile(`(?m)NOTICE: No license is installed[^\n]*\n?`)
 	out = re.ReplaceAllString(out, ``)
 
 	return out

